components/org.wso2.carbon.identity.user.endpoint/src/main/java/org/wso2/carbon/identity/user/endpoint/impl/MeApiServiceImpl.java

```diff
@@ -112,15 +117,24 @@ private Response buildSuccessfulAPIResponse(NotificationResponseBean notificatio
 
         // Check whether detailed api responses are enabled.
         if (isDetailedResponseBodyEnabled()) {
-            SuccessfulUserCreationDTO successfulUserCreationDTO = buildSuccessResponse(notificationResponseBean);
+            String notificationChannel = notificationResponseBean.getNotificationChannel();
+            if (StringUtils.isNotEmpty(notificationChannel) && notificationChannel
+                    .equals(NotificationChannels.EXTERNAL_CHANNEL.getChannelType())) {
+                // Handle response when the notifications are externally managed.
+                SuccessfulUserCreationExternalResponseDTO successfulUserCreationDTO =
+                        buildSuccessResponseForExternalChannel(notificationResponseBean);
+                return Response.status(Response.Status.CREATED).entity(successfulUserCreationDTO).build();
+            }
+            SuccessfulUserCreationDTO successfulUserCreationDTO =
+                    buildSuccessResponseForInternalChannels(notificationResponseBean);
             return Response.status(Response.Status.CREATED).entity(successfulUserCreationDTO).build();
         } else {
             if (notificationResponseBean != null) {
                 String notificationChannel = notificationResponseBean.getNotificationChannel();
                 // If the notifications are required in the form of legacy response, and notifications are externally
                 // managed, the recoveryId should be in the response as text.
                 if (StringUtils.isNotEmpty(notificationChannel) && notificationChannel
-                        .equals(Constants.EXTERNAL_NOTIFICATION_CHANNEL)) {
+                        .equals(NotificationChannels.EXTERNAL_CHANNEL.getChannelType())) {
```

Comment:
we can get rid of StringUtils.isNotEmpty(notificationChannel)  by interchaning the variable with the constant as suggested above.